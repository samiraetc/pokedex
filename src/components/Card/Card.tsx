import { Link } from 'react-router-dom';
import { Pokemon, PokemonProps } from '../../types/typing';
import { mock } from '../../utils/mock';
import Pokemons from '../Pokemon/Pokemon';

export const COMPONENT_ID = 'pokedex-card'

interface CardProps {
    pokemon: Pokemon
}

const Card = ({ pokemon }: CardProps) => {
    return (
        <div id={COMPONENT_ID}>
            <Link to={`/details/${pokemon.id}`} >
                <div className="card">
                    <Pokemons pokemon={pokemon} />
                </div>
            </Link>
        </div>
    )
}

export default Card;