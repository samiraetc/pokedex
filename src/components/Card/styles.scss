.card {
    position: relative;
    width: 250px;
    height: 230px;
    box-shadow: 0px 0px 25px -5px rgb(183 189 193 / 40%);
    border-radius: 15px;
    text-align: center;
    margin: 0 $pokedex-spacing-12 $pokedex-spacing-12 0;
    display: inline-block;
    vertical-align: middle;

    &:hover {
        -webkit-box-shadow: 0px 0px 36px -3px rgba(181, 181, 181, 1);
        -moz-box-shadow: 0px 0px 36px -3px rgba(181, 181, 181, 1);
        box-shadow: 0px 0px 36px -3px rgba(181, 181, 181, 1);
    }
}