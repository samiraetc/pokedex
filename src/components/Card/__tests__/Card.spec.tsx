import { render, configure, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import Card, { COMPONENT_ID } from '../Card';

configure({
    testIdAttribute: 'id',
  });
  
describe('[LOADING]', () => {
    test('should render Loading', () => {
        const { asFragment, getByTestId, getByText } = render(<Card pokemon={{
            name: 'raticate',
            id: 20,
            abilities: [{ ability: { name: 'run-away' } }
            ],
            sprites: { front_default: 'string' },
            types: [{ type: { name: 'normal' } }],
            stats: [{ base_stat: 20, stat: { name: 'hp' } }],
            weight: 185,
            height: 7,
        }}/>, {wrapper: BrowserRouter});

        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});