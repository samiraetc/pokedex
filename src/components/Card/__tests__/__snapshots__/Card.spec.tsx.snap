// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LOADING] should render Loading 1`] = `
<DocumentFragment>
  <div
    id="pokedex-card"
  >
    <a
      href="/details/20"
    >
      <div
        class="card"
      >
        <div
          class="pokemon"
          id="pokedex-pokemon"
        >
          <img
            alt="raticate"
            class="pokemon-avatar"
            src="string"
          />
          <div
            class="pokemon__infos"
          >
            <p
              class="pokemon__name"
            >
              raticate
            </p>
            <p
              class="pokemon__number"
            >
              #20
            </p>
          </div>
          <div
            class="pokemon__infos"
          >
            <p>
              18.5 kg
            </p>
            <p>
              0.7 m
            </p>
          </div>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
